pkg/collector/pod_collector.go

```diff
@@ -99,11 +99,14 @@ func (c *PodCollector) GetMetrics() ([]CollectedMetric, error) {
 		isPodReady, podReadyAge := GetPodReadyAge(pod)
 
 		if isPodReady {
-			if podReadyAge >= c.minPodReadyAge {
-				go c.getPodMetric(pod, ch, errCh)
-			} else {
+			if pod.DeletionTimestamp != nil {
+				skippedPodsCount++
+				c.logger.Warnf("Skipping metrics collection for pod %s/%s because it is seems to be scheduled for termination (DeletionTimestamp: %s)", pod.Namespace, pod.Name, pod.DeletionTimestamp)
```

Comment:
nit: I would make this message more clear and say: `Skipping metrics collection for pod %s/%s because it is being terminated (DeletionTimestamp: %s)`

I would also propose we just do this a Debug log to be less noisy.

Reply:
@mikkeloscar okay, agree about the message. Should set a debug level only for this kind of message of for other two too?
Maybe we can leave Warn for "podReadyAge < c.minPodReadyAge" condition check and "pod not ready ready" & "pod terminating" cases can go to the debug level?

Reply:
Sounds good to me!

Reply:
Pushed.